### PR TITLE
feature: stop commiting generated files

### PR DESCRIPTION
**Pull request by SMillerDev**
Part 2 of the conversion to GitHub actions deploys to pages

### CODE REVIEW

File: .github/workflows/scheduled.yml [hunk 1]
```diff
@@ -67,35 +62,50 @@ jobs:
...
-          git add _data/formula{,_canonical.json} _data/bottle api/formula api/bottle formula
-          if ! git diff --no-patch --exit-code HEAD -- _data/analytics _data/formula{,_canonical.json} _data/bottle api/formula api/bottle formula cask; then
-            git commit -m "formula: updates from homebrew/core" _data/analytics _data/formula{,_canonical.json} _data/bottle api/formula api/bottle formula cask
-          fi
+      - uses: actions/upload-artifact@v3
+        if: matrix.os == 'macos-latest'
+        with:
+          name: data-macos
+          path: _data/
+          retention-days: 5
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:70]: Can this retention days be lower?

File: .github/workflows/scheduled.yml [hunk 2]
```diff
@@ -67,35 +62,50 @@ jobs:
...
+          name: formula
+          path: formula/
+          retention-days: 5
 
-      - name: Push commits
-        uses: Homebrew/actions/git-try-push@master
+      - name: Update analytics data for Homebrew/linuxbrew-core
+        if: matrix.os == 'ubuntu-22.04'
+        run: rake linux_analytics
+      - uses: actions/upload-artifact@v3
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:103]:
```suggestion
        run: rake linux_analytics

      - uses: actions/upload-artifact@v3
```

File: .github/workflows/scheduled.yml [hunk 3]
```diff
@@ -116,6 +126,12 @@ jobs:
         run: |
           gem install bundler -v "~>1"
           bundle install --jobs 4 --retry 3
+      - uses: actions/download-artifact@v3
+      - name: Move artifacts into place
```
Comment (MikeMcQuaid) [inline .github/workflows/scheduled.yml:130]:
```suggestion
          bundle install --jobs 4 --retry 3

      - uses: actions/download-artifact@v3

      - name: Move artifacts into place
```

**Review by MikeMcQuaid — APPROVED** [review]
Fine as-is, a few optional nits.

File: .github/workflows/scheduled.yml (hunk 1 above)
Comment (SMillerDev) [reply .github/workflows/scheduled.yml:70]: Yeah sure, what did you have in mind?

Comment (MikeMcQuaid) [reply .github/workflows/scheduled.yml:70]: @SMillerDev I guess `1`?
```suggestion
          retention-days: 1
```

**Comment by MikeMcQuaid** [comment]
@SMillerDev how's this looking?

**Comment by SMillerDev** [comment]
I keep wanting to merge it when I have time to look after it, so far that just had me forgetting about it

**Pull request merged**